apps/svelte.dev/src/routes/docs/[...path]/+page.svelte

```diff
@@ -17,6 +20,37 @@
 		const link = 'docs/' + data.document.file.split('/').slice(2).join('/');
 		return `https://github.com/sveltejs/${name}/edit/main/documentation/${link}`;
 	});
+
+	// make hash case-insensitive
+	// hash was lowercase in v4 docs and varying case in v5 docs
+	function get_url_to_redirect_to() {
+		const hash = $page.url.hash.replace(/^#/i, '');
+		if (!hash) return;
+
+		const elements = document.querySelectorAll('*[id]');
+		// if there's an exact match, use that. no need to redirect
+		// also handles the case where one appears twice with difference casing
+		// e.g. https://svelte.dev/docs/kit/@sveltejs-kit#redirect vs https://svelte.dev/docs/kit/@sveltejs-kit#Redirect
+		for (const el of elements) {
+			if (el.id === hash) {
+				return;
+			}
+		}
+		for (const el of elements) {
+			if (el.id.toLocaleLowerCase() === hash.toLocaleLowerCase()) {
+				const url = $page.url;
+				url.hash = el.id;
```

Comment:
is it necessary to mutate it? wouldn't you be able to construct the new url and return it without touching the store?

Reply:
yeah, mostly I wanted to know if I should even be able to. It doesn't seem like I should. I filed a feature request for that: https://github.com/sveltejs/kit/issues/12870. And have updated this to make a copy